### ISSUE TITLE
(🐞) false positive expression contains `Any`, type arguments assumed to be `Any` when assigning to a variable of type `Any`

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#12147]
```py
from typing import Any

a: Any

a = {1: 2}  # error: Expression type contains "Any" (has type "Dict[Any, Any]")
```

Doesn't look to me like the expression contains `Any`.

[playground](https://mypy-play.net/?mypy=latest&python=3.10&flags=strict%2Cdisallow-any-expr&gist=f475839740d52986fee7af7e1c52ca86)

**Comment by cbscsm** [comment]
It is not a bug, I think you checked the option `--disallow-any-expr`.

![image](https://user-images.githubusercontent.com/31615733/153113757-376e0a02-f2e6-4fe3-a963-113c36d13a29.png)
